receiver/receivercreator/observerhandler.go

```diff
@@ -148,6 +150,10 @@ func (obs *observerHandler) OnRemove(removed []observer.Endpoint) {
 	defer obs.Unlock()
 
 	for _, e := range removed {
+		if env, err := e.Env(); err == nil {
+			obs.logger.Debug("handling removed endpoint", zap.Any("env", env))
+		}
```

Comment:
This looks very strange to call the func just to do logging.... Also you simply ignore the error...

Reply:
Endpoint discovery is relatively low frequency, but fair point. How does this checked entry suit you?

Reply:
looks good.